### ISSUE TITLE
Schema definition of azurerm_policy_definition object parameters does not get applied

### ISSUE THREAD

**Issue by mhennecke** [hashicorp/terraform-provider-azurerm#19720]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.3

### AzureRM Provider Version

3.34.0

### Affected Resource(s)/Data Source(s)

azurerm_policy_definition

### Terraform Configuration Files

```hcl
resource "azurerm_policy_definition" "test" {
  name         = "zztest"
  policy_type  = "Custom"
  mode         = "Indexed"
  display_name = "zztest"
  policy_rule = <<POLICY_RULE
	{
    "if": {
      "field": "location",
      "like": "[parameters('testObject').location]"
    },
    "then": {
      "effect": "audit"
    }
  }
POLICY_RULE
  parameters = <<PARAMETERS
	{
    "testObject": {
      "type": "Object",
      "metadata": {
        "description": "test",
        "displayName": "test"
      },
      "schema": {
        "description": "test",
        "type": "object",
        "properties": {
          "location": {
            "description": "test",
            "type": "string"
          }
        },
        "additionalProperties": false
      }
    }
  }
PARAMETERS
}
```

### Debug Output/Panic Output

```shell
-
```

### Expected Behaviour

The policy definition is created and the parameter `testObject` contains the `schema` information. Subsequent plans should not show any changes for the policy

### Actual Behaviour

The `terraform apply` creates the policy definition and completes successfully. However, the created policy definition is missing the `schema` information for the `testObject` parameter. Subsequent plans show an in-place update for the `testObject` parameter for adding the `schema`.

Adding the schema information manually, e.g. in the Azure portal, to the definition and running a plan/apply afterwards shows the same behaviour.

### Steps to Reproduce

1. `terraform apply`
2. Policy definition created successfully through apply
3. Check policy definition in Azure portal (schema is missing)
4. plan/apply again (schema to be added again)

**Comment by wuxu92** [comment]
Hi @mhennecke, thanks for your feedback. the current API version does not support the `schema` field in the `parameters` property. so there is no way terraform to support it but drop it in creating, even the azure portal page support it.

reference swagger definition:

https://github.com/Azure/azure-rest-api-specs/blob/bef7ff1b9b0629adbc75e19bebc755af43930d83/specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/policyDefinitions.json#L545-L548

**Comment by anlandu** [comment]
This should be fixed in https://github.com/Azure/azure-rest-api-specs/pull/24542!

**Comment by papanito** [comment]
Apparently it's fixed in the API, but seems it still does not work. So I assume a code change of the provider is required?

**Comment by wuxu92** [comment]
@papanito @mhennecke the provider has to update the API version it used. Now that specs#24542 has been merged into the main branch, I will begin the update work.

**Comment by wuxu92** [comment]
Another issue is currently blocking this feature during the migration to hashicorp/go-azure-sdk. I will provide an update here once the block has been removed.

**Comment by Adam-McGill** [comment]
Are there any updates to implementing this feature?

**Comment by pollev** [comment]
@wuxu92 I believe this migration has been completed in the meanwhile? Any chance you can merge your PR?

**Comment by wuxu92** [comment]
@pollev we are still blocked by the sdk migration, it seems the new sdk will break some existing logics, I'll take another look this week.
